### ISSUE TITLE
User login through ajax is failing 

### ISSUE THREAD

**Issue by chaitanyaVkrishna** [enricodeleo/wpClientAjaxLogin#1]
Firstly thank you for the plugin

i have been using the plugin since a week and trying to authenticate woocommerce users via ajax using your ClientAjaxLogin plugin its working fine while displaying the errors, but the user is not getting logged in when ever the login is done through ajax, i am unable to find the issue  kindly guide me regarding the issue.
the following is my html, php and jquery code 

		<form id="modal-login" method="post" class="login" role="form">

			<?php do_action( 'woocommerce_login_form_start' ); ?>

			<p class="form-group ">
				<input type="text" class="input-text email form-control" name="username" id="username" placeholder="email" value="<?php if ( ! empty( $_POST['username'] ) ) echo esc_attr( $_POST['username'] ); ?>" />
			</p>
			<p class="form-group ">
				<input class="input-text password form-control" type="password" placeholder="Password" name="password" id="password" />
			</p>

			<?php do_action( 'woocommerce_login_form' ); ?>

			<p class="form-group form-row">
				<?php wp_nonce_field( 'woocommerce-login' ); ?>
				<input type="submit" class="button" name="login" value="<?php _e( 'Login', 'unity' ); ?>" />
				<label for="rememberme" class="inline">
					<input name="rememberme" type="checkbox" id="rememberme" value="forever" /> <?php _e( 'Remember me', 'unity' ); ?>
				</label>
			</p>
			<p class="form-group lost_password">
				<a href="<?php echo esc_url( wp_lostpassword_url() ); ?>"><?php _e( 'Lost your password?', 'unity' ); ?></a>
			</p>

		<a href="#" data-toggle="modal" data-target="#regModal">
Not A Member Yet
</a>
			<?php do_action( 'woocommerce_login_form_end' ); ?>

		</form>
		
		jQuery(document).ready(function($) {
    "use strict";
 
    $('#modal-login').on('submit', function(e){
  $.ajax({
    type: "POST",
    url: ""http://zapokart.com/wp-admin/admin-ajax.php"", 
     
    xhrFields: {
        withCredentials: true
    },
    data: {
        user: "user",
        pwd: "pwd", 
        action: "clientAjaxLogin"
    },
  success: function(response) {
        var respObj = JSON.parse(response );
        if ( response == false ) {
   window.location = respObj.success;
 } else {

           $("#message").html( respObj.error );
        }
    }
});

        e.preventDefault();
    
   });
   
    });

   
    </script>
	
<?php wp_footer(); ?>

Thanks and Regards
V.Chaitanya Krishna

**Comment by enricodeleo** [comment]
Hi @chaitanyaVkrishna and thank you. This plugin is somehow old and now I think there are better ways of doing the job. But still :) 
Are you trying to login within the same domain (eg on the frontend of your website) or from a remote client (an app or different website for example)?

**Comment by chaitanyaVkrishna** [comment]
Hi enric i am trying to login within the same domain, is it meant to be done only from a remote client?
Thanks you.

**Comment by enricodeleo** [comment]
No, this is not a requirement. Are you able to check the php error log?

**Comment by chaitanyaVkrishna** [comment]
yes i have checked php log found post admin-ajax.php, but there were no errors found.

**Comment by enricodeleo** [comment]
Did you turned on php error reporting ? (https://codex.wordpress.org/WP_DEBUG) Are you console.log-ing the response from the endpoint? Does it return data?

**Comment by chaitanyaVkrishna** [comment]
Yes i just turned it on and checked the bebug log file, no errors regarding ajax login were found, and before displaying the errors on the form, i did check console and could find the error returned if the form is left empty or if invalid credentials are entered, but even if the user credentials are correct the wp_error returns the error instead of successfully redirecting to the page .

**Comment by chaitanyaVkrishna** [comment]
Its working like charm on other server installed with wordpress, thank you Enric sorry i didn't check on other server first, there might be some plugin or theme conflict on that server which is conflicting it,  thank you once again for your response.

**Comment by enricodeleo** [comment]
happy you figured out @chaitanyaVkrishna (and happy this plugin still works and is useful for people out there :))
